### DISCUSSION TITLE
Feedback

### DISCUSSION THREAD

**Discussion by UnconnectedBedna** [UnconnectedBedna/shrink-backup#1] [General]
Feel free to leave feedback.

**Comment by MeestorX** [comment]
This script is a lifesaver. Before this, I couldn't figure out an easy way to create a downloadable image that wasn't the size of the sd card. (typically 64gb!).
An essential tool for anyone distributing images or backing up their Armbian sbc. Thank you for doing this!

**Comment by CGA11** [comment]
Just wanted to share my appreciation for your efforts developing this excellent tool. Btw, are there any plans for an update supporting auto expansion on Raspberry Pi OS based on Debian Trixie?

**Reply by UnconnectedBedna** [reply]
Is that released already?
I know debian is out, but did not know rpiOS had a new official trixie release...?

And yes ofc! autoexpansion for rpiOS should work (for official releases) and I thought it would, because the rpiOS autoexpansion is "generic" for rpiOS so it _should theoretically_ work on any system.

If you have a rpiOS with trixie installed (official version, I have not yet checked if the pi foundation actually made a new release) and autoexpansion does not work, please create a feature request for the new version of rpiOS.
Otherwise I will get around to it, but it might go faster if you kick me in the butt with a feature request. xD

rpiOS autoexpansion:
https://github.com/UnconnectedBedna/shrink-backup/blob/59e27b2cb49d79bef607fd26adb3d8903e5ae9ac/shrink-backup#L2211-L2266

**Comment by CGA11** [comment]
[Indeed](https://www.phoronix.com/news/Raspberry-Pi-OS-Debian-13) it has, but I can't blame you for not knowing, it's just been out a couple of days. Anyways, no hurry, I'm not gonna kick you in the butt :smile:

**Reply by UnconnectedBedna** [reply]
But you are saying you have tried and it does not work?
(Darn I didn't find this out before, it was released 3 days ago and I could have made it work during friday-sunday, well well, hopefully it's just minor changes needed to be done anyway)

**Reply by CGA11** [reply]
Yes, I switched from Manjaro ARM (EOL?) to Raspberry OS just when Trixie was released.
``` 
## Scanning filesystem and calculating...
!! Unknown OS, no autoexpansion available

**Reply by UnconnectedBedna** [reply]
Ah, I see, it's this failing:
https://github.com/UnconnectedBedna/shrink-backup/blob/59e27b2cb49d79bef607fd26adb3d8903e5ae9ac/shrink-backup#L2593-L2597

I'll download and install the new version and figure out a way to detect if it's running rpiOS trixie...

Thank you for the heads up!

Side note: don't think I didn't notice your posts on reddit, I don't have an account so I "can't" thank you there.
Those posts were at some point about 30-40% of the incoming link traffic.
Thank you for that. :heart: :smile_cat: :heart:

**Reply by CGA11** [reply]
`cat /etc/os-release` points to Debian 13 but there's no mention of Raspberry.

**Reply by CGA11** [reply]
> Thank you for that. 

You're most welcome.

**Reply by UnconnectedBedna** [reply]
> cat /etc/os-release points to Debian 13 but there's no mention of Raspberry.

Yeah, I can't really use that for rpiOS, they are all detected as "debian", even as "pretty name" IIRC (or whatever it's called), hence me using the apt lists, but there sure is a way, I just have to keep both methods so it still works on older releases of rpiOS. :)
I also have to be careful so other OS:es are not also detected as rpi when they are not.

**Reply by UnconnectedBedna** [reply]
Lo and behold, it wasn't very complicated.
https://github.com/UnconnectedBedna/shrink-backup/commit/caaac29b2bd5a9f7b950de9344ff71eba9897c2d

Feel free to switch to testing branch and see if it works.
I still have to do some tests for other operating systems before making a full release on stable branch.

**Reply by CGA11** [reply]
Great! I'll try to give it a go during the day.

**Reply by CGA11** [reply]
Worked like a charm, thank you for the quick fix!
